beluga_amcl/test/filter_update_control/test_filter_update_control_mixin.cpp

```diff
@@ -205,10 +234,29 @@ TEST_F(FilterUpdateControlTests, ResamplingPollerWithNPoliciesUsesShortCircuitEv
     EXPECT_CALL(uut, sample()).Times(1);
     EXPECT_CALL(uut, update_sensor(_)).Times(1);
     EXPECT_CALL(uut, reweight()).Times(1);
+
+    // iteration 7
+    EXPECT_CALL(uut, update_motion(_)).Times(1);
+    EXPECT_CALL(uut, sample()).Times(1);
+    EXPECT_CALL(uut, update_sensor(_)).Times(1);
+    EXPECT_CALL(uut, reweight()).Times(1);
+
+    // iteration 8
+    // no calls on the filter
+
+    // iteration 9
+    EXPECT_CALL(uut, update_motion(_)).Times(1);
+    EXPECT_CALL(uut, sample()).Times(1);
+    EXPECT_CALL(uut, update_sensor(_)).Times(1);
+    EXPECT_CALL(uut, reweight()).Times(1);
+    EXPECT_CALL(uut, resample()).Times(1);
   }
 
+  auto next_force_update_flag_value = force_update_values.cbegin();
   for (const auto& iteration_result : expected_return_values) {
```

Comment:
@glpuga nit^2: you may want to use `ranges::views::zip` here (can't find better usage example than [this](https://stackoverflow.com/questions/61579631/ranges-v3-zip-with-auto-and-structured-bindings)).

Reply:
fixed.